paddle/fluid/operators/common_infer_shape_functions.cc

```diff
@@ -98,49 +101,46 @@ void UnaryOpUnchangedInferShapeCheckAxis(framework::InferShapeContext *ctx) {
           "Attr(axis) value should be in range [-R, R-1], "
           "R is the rank of Input(X). But received axis: %d, R: %d.",
           axis, x_rank));
-  ctx->ShareDim(ctx->GetInputNameByIdx(0), /*->*/ ctx->GetOutputNameByIdx(0));
-  ctx->ShareLoD(ctx->GetInputNameByIdx(0), /*->*/ ctx->GetOutputNameByIdx(0));
+  ctx->ShareDim(x_name, /*->*/ out_name);
+  ctx->ShareLoD(x_name, /*->*/ out_name);
 }
 
 // broadcast input(0) and input(1) -> output(0)
 void BinaryOpBroadcastInferShape(framework::InferShapeContext *ctx) {
   auto x_name = ctx->GetInputNameByIdx(0);
   auto y_name = ctx->GetInputNameByIdx(1);
   auto out_name = ctx->GetOutputNameByIdx(0);
-  OP_INOUT_CHECK(ctx->HasInput(x_name), "Input", x_name, "ElementwiseOp");
-  OP_INOUT_CHECK(ctx->HasInput(y_name), "Input", y_name, "ElementwiseOp");
-  OP_INOUT_CHECK(ctx->HasOutput(out_name), "Output", out_name, "ElementwiseOp");
   auto x_dims = ctx->GetInputDim(x_name);
```

Comment:
not important, I think we can add some comment, tell developers "why not need to check HasInput and HasOutput here"

Reply:
Thanks, I will add comments in the next PR.